homeassistant/components/sensor/recorder.py

```diff
@@ -200,11 +200,15 @@ def _normalize_states(
                     hass.data[WARN_UNSTABLE_UNIT] = set()
                 if entity_id not in hass.data[WARN_UNSTABLE_UNIT]:
                     hass.data[WARN_UNSTABLE_UNIT].add(entity_id)
+                    extra = ""
+                    if old_metadata := statistics.get_metadata(hass, entity_id):
+                        extra = f" and matches the unit of already compiled statistics {old_metadata['unit_of_measurement']}"
```

Comment:
```suggestion
                        extra = f"and matches the unit of already compiled statistics {old_metadata['unit_of_measurement']}"
```

Reply:
If we do like this, there's a useless `' '` at the end of the log message when there's no previous metadata 🤷‍♂️